File: Hijeck/app/src/main/java/com/example/hijeck/UserSecretAdapter.java
package com.example.hijeck;

import android.content.Context;
import android.os.Build;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;

import java.util.ArrayList;
import java.util.List;

public class UserSecretAdapter extends ArrayAdapter{
    private Context context;
    List list=new ArrayList();
    int res;

    public UserSecretAdapter(@NonNull Context context, int resource) {
        super(context, resource);
    }

    @Override
    public void add(Object object)
    {
        super.add(object);
        list.add(object);
    }

    public int getCount()
    {
        return list.size();
    }

    public Object getItem(int position)
    {
        return list.get(position);
    }

    static class LayoutHandler
    {
        TextView Application_name,USerNAme,Pass;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        View row=convertView;
        LayoutHandler layoutHandler;
        if(row==null)
        {
            LayoutInflater layoutInflater=(LayoutInflater)this.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            row=layoutInflater.inflate(R.layout.adapterview,parent,false);
            layoutHandler= new LayoutHandler();
            layoutHandler.Application_name=(TextView)row.findViewById(R.id.application_name_text);
            layoutHandler.USerNAme=(TextView)row.findViewById(R.id.user_name_text);
            layoutHandler.Pass=(TextView)row.findViewById(R.id.pass_text);
            row.setTag(layoutHandler);

        }
        else
            {
                layoutHandler =(LayoutHandler)row.getTag();
            }

        UserSecrets us=(UserSecrets)this.getItem(position);
        layoutHandler.Application_name.setText(us.getApplicationame());
        layoutHandler.USerNAme.setText(us.getUsername());
        layoutHandler.Pass.setText(us.getPassword());

        return  row;

    }

    

}

File: GoogleSummerOfCode/app/src/main/java/com/example/googlesummerofcode/UserSecretAdapter.kt
package com.example.googlesummerofcode

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import java.util.*

class UserSecretAdapter(context: Context, resource: Int) :
    ArrayAdapter<Any?>(context, resource) {

    var list: ArrayList<Any?> = ArrayList<Any?>()
    var res = 0
    override fun add(`object`: Any?) {
        super.add(`object`)
        list.add(`object`)
    }

    override fun getCount(): Int {
        return list.size
    }

    override fun getItem(position: Int): Any? {
        return list[position]
    }

    internal class LayoutHandler {
        var Application_name: TextView? = null
        var USerNAme: TextView? = null
        var Pass: TextView? = null
    }

    override fun getView(
        position: Int,
        convertView: View?,
        parent: ViewGroup
    ): View {
        var row = convertView
        val layoutHandler: LayoutHandler
        if (row == null) {
            val layoutInflater =
                getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            row = layoutInflater.inflate(R.layout.adapterview, parent, false)
            layoutHandler = LayoutHandler()
            layoutHandler.Application_name =
                row.findViewById<View>(R.id.application_name_text) as TextView
            layoutHandler.USerNAme =
                row.findViewById<View>(R.id.user_name_text) as TextView
            layoutHandler.Pass = row.findViewById<View>(R.id.pass_text) as TextView
            row.tag = layoutHandler
        } else {
            layoutHandler = row.tag as LayoutHandler
        }
        val us = getItem(position) as UserSecrets?
        if (us != null) {
            layoutHandler.Application_name?.setText(us.applicationame)
        }
        if (us != null) {
            layoutHandler.USerNAme?.setText(us.username)
        }
        if (us != null) {
            layoutHandler.Pass?.setText(us.password)
        }
        return row!!
    }
}